pkg/endpoint/policy.go

```diff
@@ -596,11 +617,18 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 // Called with e.Mutex UNlocked
 func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	metrics.EndpointCountRegenerating.Inc()
+	regenerateStart := time.Now()
 	defer func() {
 		metrics.EndpointCountRegenerating.Dec()
 		if retErr == nil {
 			metrics.EndpointRegenerationCount.
 				WithLabelValues(metrics.LabelValueOutcomeSuccess).Inc()
+
+			// Capture successful endpoint generation time
+			regenerateTime := time.Since(regenerateStart).Seconds() * secondsToMs
```

Comment:
Keep the second as the time unit.

Reply:
The reason I felt millisecond is more apt is because on an average, I am  generally seeing the generation times in  a 20 endpoint setup in the range of 3-4 milliseconds. With `seconds` as the granularity the prometheus metrics remain 0 till the sum will add up and give anything visible or meaningful after a very long time. I felt this is not very useful and to actually determine that the `generation time` is too off (say 200 ms) then we know for sure something is off with the endpoint/policy generation.  Please let me know if you feel this rationale is good?

Reply:
Please note that the earlier logic i.e `time.Since(regenerateStart) / time.Second` i.e  `int64 division` returns 0 if the value is in milliseconds.. Are you suggesting I should make this a float operation? What’s the rationale of using seconds?

Reply:
This is to follow Prometheus conventions: “[metrics] should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers).”
https://prometheus.io/docs/practices/naming/

Reply:
Note also that the same conventions require that you indicate the unit as a suffix: “[metrics] should have a suffix describing the unit, in plural form.” For example: `http_request_duration_seconds`.

Reply:
And yes of course you should use a float division instead of an int division.